com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -178,14 +178,7 @@ public String getSourceFileURI(String fullyQualifiedName, String sourcePath) {
         if (sourceElement instanceof IResource) {
             return getFileURI((IResource) sourceElement);
         } else if (sourceElement instanceof IClassFile) {
-            try {
-                IClassFile file = (IClassFile) sourceElement;
-                if (file.getBuffer() != null) {
-                    return getFileURI(file);
-                }
-            } catch (JavaModelException e) {
-                // do nothing.
-            }
+            return getFileURI((IClassFile) sourceElement);
```

Comment:
This change breaks most debug adapter clients finding the source.

The effect of this change is that getSourceFileURI always returns an (arguably invalid) jdt:// path now (as this method always returns a value), and so the code in StackTraceRequestHandler no longer actually checks user-specified source locations.

It's always better to use user-specified source locations (that contain the _actual source_) than returning a jdt uri which requires another server round-trip and somehow the server to (hopefully) find the source associated with the .class file.